### PR TITLE
Chain fee integration tests

### PR DESCRIPTION
**Pull request by asoliman92**
* Add FeeQuoter to SourceChain config
* Add assertions to make sure that in the `initial_deploy_test` we update the gas fees
* Clean up token assertions in `initial_deploy_test`

corresponding change on chainlink-ccip:
* https://github.com/smartcontractkit/chainlink-ccip/pull/249

### CODE REVIEW

**Review by mateusz-sekara — DISMISSED** [review]

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 1]
```diff
@@ -210,8 +218,7 @@ var SourceReaderConfig = evmrelaytypes.ChainReaderConfig{
 					ReadType:          evmrelaytypes.Method,
 				},
 				// TODO: swap with const.
-				"OnRampGetDestChainConfig": {
-					//consts.MethodNameOnRampGetDestChainConfig: {
+				consts.MethodNameOnRampGetDestChainConfig: {
```
Comment (winder) [inline core/capabilities/ccip/configs/evm/contract_reader.go:212]: TY. It should be safe to remove the "remove deprecated config" TODO's also.

**Review by winder — DISMISSED** [review]

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 2]
```diff
@@ -236,6 +243,51 @@ var SourceReaderConfig = evmrelaytypes.ChainReaderConfig{
...
+				consts.MethodNameGetValidatedTokenPrice: {
+					ChainSpecificName: mustGetMethodName("getValidatedTokenPrice", feeQuoterABI),
+					ReadType:          evmrelaytypes.Method,
+				},
+				consts.MethodNameGetFeeTokens: {
+					ChainSpecificName: mustGetMethodName("getFeeTokens", feeQuoterABI),
+					ReadType:          evmrelaytypes.Method,
+				},
+			},
+		},
```
Comment (winder) [inline core/capabilities/ccip/configs/evm/contract_reader.go]: Could you limit this to the minimum required methods? That will make things a little easier for non-EVM integrations.

Comment (asoliman92) [reply core/capabilities/ccip/configs/evm/contract_reader.go]: good point. let me do it

**Review by makramkd — DISMISSED** [review]

File: integration-tests/deployment/ccip/changeset/initial_deploy_test.go [hunk 3]
```diff
@@ -91,17 +97,126 @@ func TestInitialDeploy(t *testing.T) {
...
-		require.Equal(t, ccdeploy.MockLinkPrice, timestampedPrice.Value)
-	}
+	//for dest := range e.Chains {
+	//	linkAddress := state.Chains[dest].LinkToken.Address()
+	//	feeQuoter := state.Chains[dest].FeeQuoter
+	//	timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
+	//	require.NoError(t, err)
+	//	require.Equal(t, ccdeploy.MockLinkPrice, timestampedPrice.Value)
+	//	require.NotEqual(t, ccdeploy.InitialLinkPrice, timestampedPrice.Value)
+	//}
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/changeset/initial_deploy_test.go]: remove if not required?

File: integration-tests/deployment/ccip/changeset/initial_deploy_test.go [hunk 4]
```diff
@@ -91,17 +97,126 @@ func TestInitialDeploy(t *testing.T) {
...
+
 	// After commit is reported on all chains, token prices should be updated in FeeQuoter.
-	for dest := range e.Chains {
-		linkAddress := state.Chains[dest].LinkToken.Address()
-		feeQuoter := state.Chains[dest].FeeQuoter
-		timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
-		require.NoError(t, err)
-		require.Equal(t, ccdeploy.MockLinkPrice, timestampedPrice.Value)
-	}
+	//for dest := range e.Chains {
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/changeset/initial_deploy_test.go]: we want to remove this now then right?

File: integration-tests/deployment/ccip/changeset/initial_deploy_test.go [hunk 5]
```diff
@@ -91,17 +97,126 @@ func TestInitialDeploy(t *testing.T) {
 	// Wait for all commit reports to land.
 	ccdeploy.ConfirmCommitForAllWithExpectedSeqNums(t, e, state, expectedSeqNum, startBlocks)
 
+	// Token and Gas prices should be updated in FeeQuoter
+	assertUpdatedGas(t, e, state, initialGasUpdates)
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/changeset/initial_deploy_test.go]: Does the ConfirmCommitForAllWithExpected guarantee that the price reports would be included there? Or how do we know that the price reports will have landed by this point

Comment (asoliman92) [reply integration-tests/deployment/ccip/changeset/initial_deploy_test.go]: updating prices is part of the OffRamp commit function. so if commit happened it's safe to assume prices were updated. (unless there's a bug)

**Review by AnieeG — APPROVED** [review]
For future please create tickets for each TODO

File: integration-tests/deployment/ccip/token_info.go [hunk 6]
```diff
@@ -32,6 +33,7 @@ func (tc *TokenConfig) UpsertTokenInfo(
 func (tc *TokenConfig) GetTokenInfo(
 	lggr logger.Logger,
 	linkToken *burn_mint_erc677.BurnMintERC677,
+	wethToken *weth9.WETH9,
```
Comment (makramkd) [inline integration-tests/deployment/ccip/token_info.go:36]: I'm not super familiar with this code but is this `TokenConfig` struct really needed? We should use the existing offchain config types directly where possible to reduce this churn of code that wraps other stuff IMO. The more "glue code" there is for the tests the less likely it is devs would be effectively working on them. That's an issue with the integration-tests code today as well

File: integration-tests/deployment/ccip/test_helpers.go [hunk 7]
```diff
@@ -458,49 +467,96 @@ const (
...
+}
+
+func deploySingleFeed(
+	lggr logger.Logger,
+	ab deployment.AddressBook,
+	chain deployment.Chain,
+	deployFunc func(deployment.Chain) ContractDeploy[*aggregator_v3_interface.AggregatorV3Interface],
+	symbol TokenSymbol,
+) (common.Address, string, error) {
+	//tokenTV := deployment.NewTypeAndVersion(PriceFeed, deployment.Version1_0_0)
```
Comment (makramkd) [inline integration-tests/deployment/ccip/test_helpers.go:540]: Delete commented line

**Review by winder — APPROVED** [review]

**Review by chainchad — APPROVED** [review]

**Review by timweri — APPROVED** [review]
go.mod and go.sum files ✅

File: integration-tests/deployment/ccip/token_info.go (hunk 6 above)
Comment (asoliman92) [reply integration-tests/deployment/ccip/token_info.go:36]: It is used to initiate offchain config. Just getting the correct addresses and returning them in a correct format.

**Pull request merged**